site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,790 @@
+But the default language will be used if you don't have a translation of that language.
+Sometimes your bot might not be able to see the user's preferred language provided by their client, and in that case the default language will be used, too.
+
+**The `ctx.from.language_code` will be visible only if the user previously has started a private conversation with your bot.**
+
+<CodeGroup>
+  <CodeGroupItem title="TypeScript" active>
+
+```ts
+import { Bot, Context } from "grammy";
+import { I18n, I18nFlavor } from "@grammyjs/i18n";
+
+// For full TypeScript support and proper auto-completion
+// extend the Context with I18n's Context Flavor.
+type MyContext = Context & I18nFlavor;
+
+// Create a bot as you normally would.
+// Remember to extend the context.
+const bot = new Bot<MyContext>(""); // <-- put your bot token here (https://t.me/BotFather)
+
+// Create an `I18n` instance.
+// Continue reading to find out how to configure the instance.
+const i18n = new I18n({
+  defaultLocale: "en", // see below for more information
+
+  // Load all translation files from locales/.
+  directory: "locales",
+});
+
+// Finally, register the i18n instance in the bot,
+// so the messages get translated on their way!
+bot.use(i18n);
+
+// Everything is set up now.
+// You can access translations with `t` or `translate`.
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("start-msg"));
+});
+```
 
-But we need to set up the i18n support first as the bot needs to know where the actual text is coming from.
+</CodeGroupItem>
+ <CodeGroupItem title="JavaScript">
 
 ```js
-import { Bot, session } from "grammy";
-import { I18n } from "@grammyjs/i18n";
+const { Bot } = require("grammy");
+const { I18n } = require("@grammyjs/i18n");
 
+// Create a bot as you normally would.
+const bot = new Bot(""); // <-- put your bot token here (https://t.me/BotFather)
+
+// Create an `I18n` instance.
+// Continue reading to find out how to configure the instance.
 const i18n = new I18n({
-  defaultLanguageOnMissing: true, // implies `allowMissing` = `true`
+  defaultLocale: "en", // see below for more information
+
+  // Load all translation files from locales/.
   directory: "locales",
-  useSession: true,
 });
 
-// You can also provide i18n data directly.
-i18n.loadLocale("en", { hey: "Hello!" });
+// Finally, register the i18n instance in the bot,
+// so the messages get translated on their way!
+bot.use(i18n);
+
+// Everything is set up now.
+// You can access translations with `t` or `translate`.
+bot.command("start", async (ctx) => {
+  await ctx.reply(ctx.t("start-msg"));
+});
+```
+
+</CodeGroupItem>
+ <CodeGroupItem title="Deno">
+
+```ts
+import { Bot, Context } from "https://deno.land/x/grammy/mod.ts";
+import { I18n, I18nFlavor } from "https://deno.land/x/grammy_i18n/mod.ts";
+
+// For TypeScript and auto-completion support,
+// extend the context with I18n's flavor:
```

Comment:
```suggestion
// For full TypeScript support and proper auto-completion
// extend the Context with I18n's Context Flavor.
```
Match comment from the TypeScript example (line 162).

Reply:
The opposite was [suggested](https://github.com/grammyjs/website/pull/431#discussion_r918861264) a while ago. And I only applied it on Deno's part. My bad. I'll make everything the same.